layers/+chat/circe/packages.el

```diff
@@ -0,0 +1,162 @@
+;;; packages.el --- circe layer packages file for Spacemacs.
+;;
+;; Copyright (c) 2012-2017 Sylvain Benner & Contributors
+;;
+;; Author: Chris Barrett <chris.d.barrett@me.com>
+;; Author: Aidan Nyquist <contact@aidannyquist.com>
+;; URL: https://github.com/syl20bnr/spacemacs
+;;
+;; This file is not part of GNU Emacs.
+;;
+;;; License: GPLv3
+
+;;; Code:
+
+(defconst circe-packages
+  '(circe
+    persp-mode
+    circe-notifications
+    helm
+    helm-circe
+    ))
+
+(defun circe/init-circe ()
+  (use-package circe
+    :commands circe
+    :config
+    (progn
+      (set-face-background 'circe-prompt-face nil)
+      (setq circe-reduce-lurker-spam t)
+      (setq circe-use-cycle-completion t)
+      (setq circe-active-users-timeout (* 60 30)) ; 30 minutes
+      (setq circe-format-say "{nick}> {body}")
+      (setq circe-format-self-say (concat (propertize ">>>" 'face 'circe-self-say-face) " {body}"))
+      (setq circe-prompt-string (concat (propertize ">>>" 'face 'circe-prompt-face) " "))
+      (setq circe-highlight-nick-type 'all)
+
+      ;; Keybindings
+      (spacemacs/set-leader-keys "aic" 'circe)
+
+      (dolist (mode circe-modes)
+        (progn
+        (spacemacs/declare-prefix-for-mode mode "mc" "command")
+        (spacemacs/set-leader-keys-for-major-mode mode
+          "r" 'circe-reconnect
+          "R" 'circe-reconnect-all
+          "v" 'circe-version
+          "c n" 'circe-command-NAMES
+          "c q" 'circe-command-QUERY
+          "c h" 'circe-command-HELP
+          "c t" 'circe-command-CHTOPIC 
+          "c s" 'circe-command-SAY
+          )))
+
+      ;; Show channel name in prompt.
+
+      (defun circe/bufname-to-channame (s)
```

Comment:
use lambda instead of defun for functions you only need once. If you want these function permanently add them to funcs.el.